### PR TITLE
Use @property decorator for properties

### LINKED ISSUE #1563: Use @property as a decorator
A bunch of properties in the codebase are defined using `property` without the decorator syntax, which, while valid, isn't very intuitive anymore.

```python
def _get_some_value(self):
    ...

def _set_some_value(self, value):
    ...

some_value = property(_get_some_value, _set_some_value, doc="...")
del _get_some_value, _set_some_value
```

```python
@property
def some_value(self):
    """..."""
    ...

@some_value.setter
def some_value(self, value):
    ...
```

### PR DESCRIPTION
**Pull request by gokcegrbl**
Use @property decorator for properties

Fixes #1563

### CODE REVIEW

File: src/werkzeug/test.py [hunk 1]
```diff
@@ -467,162 +467,147 @@ def base_url(self, value):
...
-        _set_content_length,
-        doc="""The content length as integer.  Reflected from and to the
-        :attr:`headers`.  Do not set if you set :attr:`files` or
-        :attr:`form` for auto detection.""",
-    )
-    del _get_content_length, _set_content_length
-
-    def form_property(name, storage, doc):  # noqa: B902
-        key = "_" + name
-
```
Comment (lordmauve) [inline src/werkzeug/test.py]: Here you're flattening out a function that returns a descriptor and is invoked twice into duplicated copies. This is increasing code duplication, although eliminating a level of abstraction.

I would err on the side of not replacing this one.

File: src/werkzeug/test.py [hunk 2]
```diff
@@ -634,7 +619,7 @@ def server_port(self):
         pieces = self.host.split(":", 1)
         if len(pieces) == 2 and pieces[1].isdigit():
             return int(pieces[1])
-        elif self.url_scheme == "https":
+        if self.url_scheme == "https":
```
Comment (lordmauve) [inline src/werkzeug/test.py]: This (and an earlier place) is a non functional change of an `elif` into an `if`. Did a style checker tell you to do this?

File: src/werkzeug/test.py (hunk 1 above)
Comment (davidism) [reply src/werkzeug/test.py]: Good catch, but after looking at both I think I prefer the new version. I'll add a comment to both that they're related, but I don't expect these properties to change much.

File: src/werkzeug/test.py (hunk 2 above)
Comment (davidism) [reply src/werkzeug/test.py]: I think flake8 might be calling these out. Change seems fine.

File: src/werkzeug/test.py (hunk 1 above)
Comment (davidism) [reply src/werkzeug/test.py]: I think I prefer a hybrid of both approaches, I'll extract the behavior to some common private methods.

**Pull request merged**